app/ui-react/packages/api/src/WithVirtualizationHelpers.tsx

```diff
@@ -111,6 +122,49 @@ export class WithVirtualizationHelpersWrapped extends React.Component<
     return Promise.resolve();
   }
 
+  /**
+   * Unpublish the Service VDB with the specified name.
+   * @param virtualizationName the name of the virtualization associated with the service
+   * @param sqlQuery the sql query statement to execute against the virtualization
+   * @param rowlimit limit to number of rows to return
+   * @param rowOffset number of data rows to filter from the beginning of the result set
+   */
+  public async queryVirtualization(
+    virtualizationName: string,
+    sqlQuery: string,
+    rowlimit: number,
+    rowOffset: number
+  ): Promise<QueryResults> {
+    // The payload for the rest call
+    const queryBody = {
+      limit: rowlimit,
+      offset: rowOffset,
+      query: sqlQuery,
+      target: virtualizationName,
+    };
+
+    // The query results for the rest call
+    const queryResults = {
+      columns: [],
+      rows: [],
+    };
+
+    const response = await callFetch({
+      body: queryBody,
+      headers: {},
+      method: 'POST',
+      url: `${this.props.dvApiUri}metadata/query`,
+    });
+
+    if (!response.ok) {
+      throw new Error(response.statusText);
+    }
+
+    return !virtualizationName
```

Comment:
since virtualizationName must be supplied, I don't think the first condition will ever happen?

Reply:
see below